spec/ruby/core/kernel/iterator_spec.rb

```diff
@@ -1,12 +1,14 @@
 require_relative '../../spec_helper'
 require_relative 'fixtures/classes'
 
-describe "Kernel#iterator?" do
-  it "is a private method" do
-    Kernel.should have_private_instance_method(:iterator?)
+ruby_version_is ""..."2.8" do
```

Comment:
We should keep running the specs since the method is not yet removed (even though the specs here are not very good).
If there are warnings, `suppress_warning { }` can be used.